### DISCUSSION TITLE
Provide before and after item data snapshots in post update hooks

### DISCUSSION THREAD

**Discussion by mdocter** [directus/directus#3048] [Feature Requests]
After update of a collection item, it would be very convenient to have a snapshot of the old item data before the update took place. If I remember correctly, currently only a snapshot of the new full item is provided and the field(s) that changed with their new value.
In most post update cases my code needs to know if a certain field changed from value X to value Y. If so, do something or else do nothing.  

Something like Hasura provides in the `event.data.old` and `event.data.new` properties of their event trigger payload:

https://hasura.io/docs/1.0/graphql/core/event-triggers/payload.html

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
